common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -48,59 +50,21 @@
-                                           final String requestUrl,
-                                           final String redirectUri,
-                                           final HashMap<String, String> requestHeaders,
-                                           final AuthorizationAgent authorizationAgent,
-                                           final boolean webViewZoomEnabled,
-                                           final boolean webViewZoomControlsEnabled) {
-        Intent intent;
-        if (ProcessUtil.isBrokerProcess(context)) {
-            intent = new Intent(context, BrokerAuthorizationActivity.class);
-        } else {
-            intent = new Intent(context, AuthorizationActivity.class) {
-
-                @Override
-                public Object clone() {
-                    return super.clone();
-                }
-            };
-        }
-
-        intent.putExtra(AUTH_INTENT, authIntent);
-        intent.putExtra(REQUEST_URL, requestUrl);
-        intent.putExtra(REDIRECT_URI, redirectUri);
-        intent.putExtra(REQUEST_HEADERS, requestHeaders);
-        intent.putExtra(AUTHORIZATION_AGENT, authorizationAgent);
-        intent.putExtra(WEB_VIEW_ZOOM_CONTROLS_ENABLED, webViewZoomControlsEnabled);
-        intent.putExtra(WEB_VIEW_ZOOM_ENABLED, webViewZoomEnabled);
-        intent.putExtra(DiagnosticContext.CORRELATION_ID, DiagnosticContext.getRequestContext().get(DiagnosticContext.CORRELATION_ID));
-        return intent;
-    }
+    public static final String TAG = AuthorizationActivity.class.getSimpleName();
 
-    public static AuthorizationFragment getAuthorizationFragmentFromStartIntent(@NonNull final Intent intent) {
-        AuthorizationFragment fragment;
-        final AuthorizationAgent authorizationAgent = (AuthorizationAgent) intent.getSerializableExtra(AUTHORIZATION_AGENT);
-        Telemetry.emit(new UiStartEvent().putUserAgent(authorizationAgent));
-
-        if (authorizationAgent == AuthorizationAgent.WEBVIEW) {
-            fragment = new WebViewAuthorizationFragment();
-        } else {
-            fragment = new BrowserAuthorizationFragment();
-        }
-
-        fragment.setInstanceState(intent.getExtras());
-        return fragment;
-    }
+    private AuthorizationFragment mFragment;
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        mFragment = getAuthorizationFragmentFromStartIntent(getIntent());
+        Fragment fragment = AuthorizationActivityFactory.getAuthorizationFragmentFromStartIntent(getIntent());
+        if(fragment instanceof AuthorizationFragment){
+            mFragment = (AuthorizationFragment)fragment;
+            mFragment.setInstanceState(getIntent().getExtras());
+        }else{
+            IllegalStateException ex = new IllegalStateException("Unexpected fragment type.");
+            Logger.error(TAG, "Did not receive AuthorizationFragment from factory", ex);
```

Comment:
What might cause this?

Reply:
A bug in the factory returning something unexpected.